File: publicq-main/src/PublicQ.Application/Models/Pages/PageDto.cs
using System.Text.Json;

namespace PublicQ.Application.Models.Pages;

/// <summary>
/// Page Data Transfer Object
/// </summary>
public class PageDto
{
    /// <summary>
    /// Page type
    /// <see cref="PageType"/>
    /// </summary>
    public PageType Type { get; set; }
    
    /// <summary>
    /// Page title
    /// </summary>
    public string Title { get; set; } = string.Empty;
    
    /// <summary>
    /// Page body
    /// </summary>
    public string Body { get; set; } = string.Empty;
    
    /// <summary>
    /// Page JSON data
    /// Can be used to store additional information related to the page
    /// </summary>
    public string JsonData { get; set; } = string.Empty;
    
    /// <summary>
    /// When the entity was created (in UTC)
    /// </summary>
    public DateTime? CreatedAtUtc { get; set; }
    
    /// <summary>
    /// Created by user identifier
    /// </summary>
    public string CreatedBy { get; set; } = string.Empty;
    
    /// <summary>
    /// Updated by user identifier
    /// </summary>
    public string? UpdatedBy { get; set; }
    
    /// <summary>
    /// Update timestamp (in UTC)
    /// </summary>
    public DateTime? UpdatedAtUtc { get; set; }
}

File: publicq-main/client/src/models/page-dto.ts
import { PageType } from './page-type';

/**
 * Page Data Transfer Object
 */
export interface PageDto {
  /**
   * Unique identifier
   */
  id: string;
  
  /**
   * Page type
   */
  type: PageType;
  
  /**
   * Page title
   */
  title: string;
  
  /**
   * Page body
   */
  body: string;
  
  /**
   * Page JSON data
   * Can be used to store additional information related to the page
   */
  jsonData: string;
  
  /**
   * When the entity was created (in UTC)
   */
  createdAtUtc: string;
  
  /**
   * Created by user identifier
   */
  createdBy: string;
  
  /**
   * Updated by user identifier
   */
  updatedBy?: string | null;
  
  /**
   * When the entity was last updated (in UTC)
   */
  updatedAtUtc?: string | null;
}

